### PR TITLE
🎉  New destination connector for Xata.io

### PR DESCRIPTION
**Pull request by philkra**
## What
This PR adds a destination connector for ingesting records into Xata.io.

## How
- Used airbyte's templating engine for python 😄 
- used the [xata](https://pypi.org/project/xata/) package for ingesting data & bulk processing.

## Recommended reading order
1. `docs/integrations/destinations/xata.md`
2. `destination.py`

## 🚨 User Impact 🚨
no breaking change

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

### Community member or Airbyter

- [x] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [x] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [x] Connector version is set to `0.0.1`
    - [x] `Dockerfile` has version `0.0.1`
- [ ] Documentation updated
    - [x] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [x] `docs/integrations/<source or destination>/<name>.md` including changelog with an entry for the initial version. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [x] `airbyte-integrations/builds.md`
- [x] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

### CODE REVIEW

**Comment by philkra** [comment]
> Connector's bootstrap.md. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)

FYI, the link points to a `404`

**Comment by marcosmarxm** [comment]
@philkra are you able to run the integration test?

**Comment by philkra** [comment]
@marcosmarxm , yes. Apologies, I forgot to upload the screenshot.

Screenshot 2023-03-24 at 13 28 16

**Comment by alexfrancoeur** [comment]
@timroes 👋  sup

**Comment by timroes** [comment]
@alexfrancoeur Uh la la, love to see this :heart:

**Comment by alexfrancoeur** [comment]
@flash1293 @chandlerprall 👋 

(and now back to the actual PR 😜)

**Comment by philkra** [comment]
@marcosmarxm how can I help to move this PR forward?

**Comment by philkra** [comment]
Bumped the version of the Xata SDK dependency. Attached screenshot of the integration tests passing:
Screenshot 2023-04-19 at 08 36 54

**Comment by philkra** [comment]
@marcosmarxm could you please assist with the following topics in order to progress on the PR:

> - [ ] Secrets in the connector's spec are annotated with `airbyte_secret`

I can't find any references on how to add the secrets. Is this a necessary step?

> - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)

The link does not work, [comment](https://github.com/airbytehq/airbyte/pull/24192/#issuecomment-1476482790)

> - [ ] `docs/integrations/README.md`

I'm not certain what to do here. The connector is `alpha`

**Comment by alexfrancoeur** [comment]
@marcosmarxm thank you for all of your support so far! We'd love to get this integration over the line. @philkra asked a few questions, is there any chance we could get some guidance in order to move this forward? We sincerely appreciate any help you can provide 🙏

**Comment by marcosmarxm** [comment]
> 

Sorry the ghosting here @philkra and @alexfrancoeur. I'll try to get to this today with a quick feedback and ask to the team to review in next week.

File: airbyte-integrations/connectors/destination-xata/destination_xata/destination.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+            if message.type == Type.RECORD:
+                # Put record to processing queue
+                bp.put_record(message.record.stream, message.record.data)
+                count += 1
+            if message.type == Type.STATE:
+                yield message
+        bp.flush_queue()
+        print(f"total: {count}")
+        print(bp.get_stats())
+        if count != bp.get_stats()["total"] or bp.get_stats()["failed_batches"] != 0:
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/destination-xata/destination_xata/destination.py:51]: Please move the print stmt to self.logger.info

File: airbyte-integrations/connectors/destination-xata/unit_tests/unit_test.py [hunk 2]
```diff
@@ -0,0 +1,7 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+
+def test_example_method():
+    assert True
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/destination-xata/unit_tests/unit_test.py:7]: @philkra can we add some unit test to validate the functions you created?

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Some comments! Thanks for the contribution I think it's a good start for the connector.

File: airbyte-integrations/connectors/destination-xata/destination_xata/destination.py (hunk 1 above)
Comment (philkra) [reply airbyte-integrations/connectors/destination-xata/destination_xata/destination.py]: done via https://github.com/airbytehq/airbyte/pull/24192/commits/ce028d18cc155be6b7e00ecfca69ba341a4cb9fb

File: airbyte-integrations/connectors/destination-xata/unit_tests/unit_test.py (hunk 2 above)
Comment (philkra) [reply airbyte-integrations/connectors/destination-xata/unit_tests/unit_test.py]: unit tests added

**Review by marcosmarxm — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/destination-xata/integration_tests/integration_test.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+            "str_col": "example",
+            "int_col": 1,
+        }, emitted_at=0)
+    )]
+    dest = DestinationXata()
+    dest.write(
+        config=config, 
+        configured_catalog=test_stream,
+        input_messages=records
+    )
```
[the remark this answers is not in the document]
Comment (philkra) [reply airbyte-integrations/connectors/destination-xata/integration_tests/integration_test.py:80]: @marcosmarxm, without the proper credentials (`api_key`, `db_url`) in the `secrets/config.js` the test is going to fail. See, the screenshot attached, I'm able to run the test with the credentials in place.

Screenshot 2023-05-24 at 17 26 02

[the remark this answers is not in the document]
Comment (marcosmarxm) [reply airbyte-integrations/connectors/destination-xata/integration_tests/integration_test.py:80]: This is not the problem. I have the credentials configured. The problem is the write function is not working properly.
Rigth now is saying PASSED but no data is populated in Xata.io

**Review by marcosmarxm — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/destination-xata/integration_tests/integration_test.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (marcosmarxm) [reply airbyte-integrations/connectors/destination-xata/integration_tests/integration_test.py:80]: I tested the connector with Hubspot and its not working.
image

[the remark this answers is not in the document]
Comment (philkra) [reply airbyte-integrations/connectors/destination-xata/integration_tests/integration_test.py]: There was no assertion in the test. Please refer to https://github.com/airbytehq/airbyte/pull/24192/commits/36992b3896607b992c5c76835ef54847e50a03a4 to see the operation failing and asserting an exception.

[the remark this answers is not in the document]
Comment (philkra) [reply airbyte-integrations/connectors/destination-xata/integration_tests/integration_test.py]: > I tested the connector with Hubspot and its not working. image

Does the table in Xata match the shape of data from HubSpot?

**Comment by philkra** [comment]
@marcosmarxm how I can add the icon mentioned in the [test](https://github.com/airbytehq/airbyte/actions/runs/5082313222/jobs/9131916218#step:15:34) and what other tasks are missing to merge this PR?

**Comment by philkra** [comment]
ping @marcosmarxm

**Comment by alexfrancoeur** [comment]
heya @marcosmarxm, thanks again for all your help so far! We're so close to getting this over the line, is there anything else you need from us? @philkra had one question about an icon.

**Comment by marcosmarxm** [comment]
Sorry the delay to return to the contribution @philkra and thank you for all the patience. I'm try do my best to review again tomorrow.

**Comment by alexfrancoeur** [comment]
Hey Tim, of course! We actually just published this brand page, all our
logo assets should be here. Let me know if you need anything else. Thanks
for the quick follow up 😊

https://xata.io/brand

**Comment by timroes** [comment]
@philkra I've added the Xata logo to the platform (where they at the moment live) in https://github.com/airbytehq/airbyte-platform/commit/0a8b27b6bda448b84b80fe1c8b0469165fb6a780 You should be fine using now `xata.svg` as you do in your specification.

**Comment by philkra** [comment]
Awesome thanks @timroes for adding the logo. I don't know where to reference the logo though, I asked a couple of times 🤷‍♂️ Can you, please point me to the files I need to change or guides?

**Comment by timroes** [comment]
> Awesome thanks @timroes for adding the logo. I don't know where to reference the logo though, I asked a couple of times man_shrugging Can you, please point me to the files I need to change or guides?

@philkra You already linked it correctly here: https://github.com/airbytehq/airbyte/pull/24192/files#diff-cd7f7327b2a8abb8dd28ab593f20d184b21acde010488a4eb991e9dac5e8b5a0R13 This will now work as expected.

**Review by marcosmarxm — APPROVED** [review]

**Comment by philkra** [comment]
Thank you to everyone that contributed ❤️

**Comment by alexfrancoeur** [comment]
Thank you so much @timroes @philkra @marcosmarxm for all of your help here! ❤️ 

I'm curious, we'll plan to blog about this, add to our docs and do a bit of social promotion. Is there anyone on the airbyte side we could discuss co-marketing with? I'm not sure if there's a process for integrations outside the implementations. I can do some digging later today as well.

**Comment by marcosmarxm** [comment]
> Thank you so much @timroes @philkra @marcosmarxm for all of your help here! ❤️
> 
> I'm curious, we'll plan to blog about this, add to our docs and do a bit of social promotion. Is there anyone on the airbyte side we could discuss co-marketing with? I'm not sure if there's a process for integrations outside the implementations. I can do some digging later today as well.

@justinbchau can you help here?

**Comment by alexfrancoeur** [comment]
@justinbchau we'd love to collaborate on any co-marketing activities if you have the bandwidth. Streams, videos, blogs, social, etc. We're flexible. If not, you'll certainly be seeing some content on our end 😄

**Pull request merged**